build.assets/tooling/cmd/notarize-apple-binaries/gonConfig.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"debug/macho"
+	"encoding/binary"
+	"flag"
+	"io"
+	"os"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+type GonConfig struct {
+	AppleUsername string
+	ApplePassword string
+	BinaryPaths   []string
+}
+
+func NewGonConfig() *GonConfig {
+	gc := &GonConfig{}
+	flag.StringVar(&gc.AppleUsername, "apple-username", "", "Apple Connect username used for notarization")
```

Comment:
Hmm, this doesn't seem like the right place to declare flags.

I would declare all of the flags in main. Something like:

```
func main() {
   var gc GonConfig
   flag.StringVar(&gc.AppleUsername, "apple-username", ...)
   flag.Parse()
   // use gc.AppleUsername..
}
```

Reply:
Yeah, I'd move this to `args.go` and plumb the username through as an arg to `NewGonConfig()`

Reply:
I've updated this by moving the logic defining the flags and validating them to `args.go`, and the fields to `GonWrapper` stuct. @zmb3 @tcsc do these changes properly answer this issue?

Reply:
Updated to use kingpin per internal conversation